be/src/olap/schema_change.cpp

```diff
@@ -194,6 +194,64 @@ ColumnMapping* RowBlockChanger::get_mutable_column_mapping(size_t column_index)
         break; \
     }
 
+struct ConvertTypeMapHash {
+    size_t operator()(const std::pair<FieldType, FieldType>& pair) const {
+        return (pair.first + 31) ^ pair.second;
+    }
+};
+
+class ConvertTypeResolver {
+DECLARE_SINGLETON(ConvertTypeResolver);
+public:
+    bool get_convert_type_info(const FieldType from_type, const FieldType to_type) const {
+        return _convert_type_set.find(std::make_pair(from_type, to_type)) != _convert_type_set.end();
+    }
+
+    template<FieldType from_type, FieldType to_type>
+    void add_convert_type_mapping() {
+        _convert_type_set.emplace(std::make_pair(from_type, to_type));
+    }
+
+private:
+    typedef std::pair<FieldType, FieldType> convert_type_pair;
+    std::unordered_set<convert_type_pair, ConvertTypeMapHash> _convert_type_set;
+
+    DISALLOW_COPY_AND_ASSIGN(ConvertTypeResolver);
+};
+
+ConvertTypeResolver::ConvertTypeResolver() {
+    // supported type convert should annotate in doc:
+    // http://doris.incubator.apache.org/master/zh-CN/sql-reference/sql-statements/Data%20Definition/ALTER%20TABLE.html#description
+    // from varchar type
+    add_convert_type_mapping<OLAP_FIELD_TYPE_VARCHAR, OLAP_FIELD_TYPE_TINYINT>();
+    add_convert_type_mapping<OLAP_FIELD_TYPE_VARCHAR, OLAP_FIELD_TYPE_SMALLINT>();
+    add_convert_type_mapping<OLAP_FIELD_TYPE_VARCHAR, OLAP_FIELD_TYPE_INT>();
+    add_convert_type_mapping<OLAP_FIELD_TYPE_VARCHAR, OLAP_FIELD_TYPE_BIGINT>();
+    add_convert_type_mapping<OLAP_FIELD_TYPE_VARCHAR, OLAP_FIELD_TYPE_LARGEINT>();
+    add_convert_type_mapping<OLAP_FIELD_TYPE_VARCHAR, OLAP_FIELD_TYPE_FLOAT>();
```

Comment:
is VARCHAR to DOUBLE and DATETIME support?

Reply:
VAHRCHAR->DOUBLE is supported. Support VARCHAR->DATETIME maybe supported in next PR.